primitives/application-crypto/Cargo.toml

```diff
@@ -18,22 +18,31 @@ targets = ["x86_64-unknown-linux-gnu"]
 sp-core = { version = "7.0.0", default-features = false, path = "../core" }
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
-serde = { version = "1.0.136", optional = true, features = ["derive"] }
+serde = { version = "1.0.136",  default-features = false, optional = true, features = ["derive"] }
 sp-std = { version = "5.0.0", default-features = false, path = "../std" }
 sp-io = { version = "7.0.0", default-features = false, path = "../io" }
 
 [features]
 default = [ "std" ]
 std = [
 	"full_crypto",
+	"serde_full",
 	"sp-core/std",
 	"codec/std",
 	"scale-info/std",
-	"serde",
+	"serde/std",
```

Comment:
```suggestion
	"serde/std",
	"serde",
```

Reply:
To my understanding, with the equally named serde feature, simply importing "serde/std" should be enough.

Reply:
Yeah, I was not sure. Did you tried it?

Reply:
I think both should be required? We maybe need to do `dep:serde/std` again?

Reply:
> Yeah, I was not sure. Did you tried it?

Yes, I tested it. I did not find any detailed doc that actually describes how to import features with `dep: ..`, but to be sure, I tested it with a simple project: https://github.com/haerdib/cargo-features.

It only compiles if serde implements the `std` feature. Removing `std` from `serde` (here: https://github.com/haerdib/cargo-features/blob/master/Cargo.toml#L14-L16) results in a compile error. This shows that the feature `serde` as well as the `std` part of serde are imported if `serde/std` is included.

Reply:
I stuck my nose here and cross-checked that, seems that `serde/std` should be enough. `cargo metadata` shows that `serde` features were enabled in `sp-core` and `scale-info`.

btw: test proves that `std` for `serde` was enabled. It does not check that additional features were enabled in dependent creates, e.g.:
```
 19 serde = [
 20         "dep:serde",
 21         "some-crate/some-feature"
 22 ]
```
I've added that locally and it works fine in test too.

Reply:
Yes, that's true. Thanks a lot for confirming this 👍